apps/find-authors-ids.py

```diff
@@ -0,0 +1,284 @@
+        )
+        page["#idstatus" + clear_id].clear()
+        if confirmed:
+            page["#idstatus" + clear_id].print("ID Included")
+        else:
+            page["#idstatus" + clear_id].print("ID Excluded")
+
+    # Verify if the link is already linked to the author, included or excluded, with the same type.
+    def is_linked(link, type, author_name):
+        already_linked = get_links(type, author_name)
+        strippedlink = link[1:] if link.startswith("!") else link
+        for links in already_linked:
+            if strippedlink in links:
+                return links
+        return False
+
+    def get_links(type, author_name):
+        author = get_authors(author_name)[0]
+        links = []
+        for link in author.links:
+            if link.type == type:
+                links.append(link.link)
+        return links
+
+    def get_authors(name):
+        authors = []
+        stmt = select(sch.Author).filter(sch.Author.name.like(f"%{name}%"))
+        try:
+            results = list(db.session.execute(stmt))
+            for (r,) in results:
+                authors.append(r)
+        except Exception as e:
+            print("Error : ", e)
+        return authors
+
+    def get_buttons(auth, author_id, link, included=None):
+        includeButton = H.button["button"](
+            "Include ID",
+            onclick=(
+                lambda event, auth=auth, author_id=author_id: confirm_id(
+                    auth, 1, author_id
+                )
+            ),
+        )
+        excludeButton = H.button["button", "invalidate"](
+            "Exclude ID",
+            onclick=(
+                lambda event, auth=auth, author_id=author_id: confirm_id(
+                    auth, 0, author_id
+                )
+            ),
+        )
+        if included is not None:
+            if included:
+                includeButton = H.button["button", "notavailable"](
+                    "Include ID",
+                )
+            else:
+                excludeButton = H.button["button", "notavailable"](
+                    "Exclude ID",
+                )
+        buttons = [includeButton, excludeButton]
+        return buttons
+
+    with load_config(os.environ["PAPERONI_CONFIG"]) as cfg:
+        with cfg.database as db:
+            reaserchers = get_authors(author_name)
+            author_id = reaserchers[0].author_id
+            tabIDS = []
+
+            results = prepare(
+                researchers=reaserchers,
+                idtype="semantic_scholar",
+                query_name=ss.author_with_papers,
+            )
+            async for auth, result in results:
+                link = auth.links[0].link
+                if link not in tabIDS:
+                    tabIDS.append(link)
+                    author_name_area = H.div["authornamearea"](
+                        auth.name,
+                        H.br,
+                        H.br,
```

Comment:
It would be better to handle spacing using styling (e.g. margin-top or margin-bottom on the appropriate elements).

Reply:
Fixed using `H.p` for information(name, link type), which automatically manages its spacing in [e5443de](https://github.com/mila-iqia/paperoni/pull/23/commits/e5443deae316e6b6bc180976d4c67fef3c80ac44)